local/build-binary

```diff
@@ -61,6 +62,38 @@ fi
 
 wget_loop "googletest-release-${GMOCK_VERSION}.zip" "https://github.com/google/googletest/archive/release-${GMOCK_VERSION}.zip"
 
+if [[ ${WITH_BORINGSSL} == "true" ]]; then
+    # ------------------------------------------------------------------------------
+    # compile boringssl libs
+    # ------------------------------------------------------------------------------
+    pushd ${WORKDIR}
+    git clone https://github.com/google/boringssl.git boringssl_sources
+    mkdir boringssl_sources/bin
+    cd boringssl_sources/bin
+    if [[ -f /etc/redhat-release ]]; then
+        RHVER="$(rpm --eval %rhel)"
+        case "${RHVER}" in
+            7|8)
+                if [[ ${RHVER} == 7 ]]; then
+                    source /opt/rh/go-toolset-7/enable
+                fi
+                source /opt/rh/devtoolset-8/enable
+                BORING_SSL_CMAKE=/usr/bin/cmake3
+            ;;
+            6) echo "Unsupported CentOS version" && exit 1 ;;
+        esac
+    else
+        BORING_SSL_CMAKE=/usr/bin/cmake
+    fi
+    ${BORING_SSL_CMAKE} .. -DCMAKE_CXX_FLAGS="-fPIC" -DCMAKE_C_FLAGS="-fPIC"
```

Comment:
Please, make sure that directly assigning `CMAKE_CXX_FLAGS` and `CMAKE_C_FLAGS` does not overwrite other flags.
To be honest well-written CMake should handle this case properly, but MySQL/Percona Server, for instance, does not.

Reply:
Would such construction be sufficient for this purpose?
```
    BORING_SSL_FLAGS="-fPIC"
    if ${BORING_SSL_CMAKE} .. -DCMAKE_CXX_FLAGS="${BORING_SSL_FLAGS}" -DCMAKE_C_FLAGS="${BORING_SSL_FLAGS}"; then
        unset BORING_SSL_FLAGS
    fi
```

Reply:
I double checked with the BoringSSL source code.
Here
https://github.com/google/boringssl/blob/master/CMakeLists.txt#L153
I believe everything is fine, according to this line final `CMAKE_CXX_FLAGS` is formed based on the value passed via command line options. Only then custom compiler options from the cmake file are appended.
So, no changes required here. Initial `${BORING_SSL_CMAKE} .. -DCMAKE_CXX_FLAGS="-fPIC" -DCMAKE_C_FLAGS="-fPIC"` is fine.

Reply:
Understood, thank you